xarray/core/dataset.py

```diff
@@ -2416,6 +2416,60 @@ def from_dataframe(cls, dataframe):
             obj[name] = (dims, data)
         return obj
 
+    def to_dask_dataframe(self, set_index=False):
+        """
+        Convert this dataset into a dask.dataframe.DataFrame.
+
+        Both the data and coordinate variables in this dataset form the columns of the
+        DataFrame.
+
+        If set_index=True, the dask DataFrame is indexed by
+        this dataset's coordinate. Since dask DataFrames to not support
+        multi-indexes, set_index only works if there is one coordinate dimension.
+        """
+
+        import dask.dataframe as dd
+
+        ordered_dims = self.dims
+        coord_columns = [c for c in self if c in ordered_dims]
+
+        # convert all coordinates into data variables
+        ds = self.reset_index(coord_columns).reset_coords()
```

Comment:
Is this actually necessary? Everything is already in `._variables`.

Reply:
My intent was to be able to call Dataset.chunk() so that all of the variables (both coordinates and data) were first converted to dask arrays. But IndexVariables only have a dummy .chunk() method.  

I'll change it so coordinate variables are converted with .to_base_variable()